### DISCUSSION TITLE
Azure OpenAI and OpenAI dependencies

### DISCUSSION THREAD

**Discussion by JonathanGiles** [spring-projects/spring-ai#54] [General]
Hi all / @markpollack - I'm pleased to see OpenAI support for Azure and direct-to-OpenAI offerings. One thing I wanted to note in case you were unaware - the Azure OpenAI library support both offerings, and there is no need to use a separate library for each offering. It might make your development simpler.

* To connect to Azure OpenAI: 
   * Set Azure key from portal (using the KeyCredential)
   * Set the Azure OpenAI resource endpoint (required)

* Open AI
   * Set the OpenAI key that is available on OpenAI portal (using the KeyCredential)
   * Setting the endpoint is optional

**Comment by markpollack** [comment]
I understand, and I may have already replied to you in github issues.  At the time of starting this I needed at least two implementations to make sure the AIClient interface was generic.  There is a hugging face support on a branch that I need to merge.  I'm sort of fine to keep both implementation options, but highlight it could be used for both.  I've seen some differences in terms of metadata returned for inference calls that I'm not sure how to reconcile.
